code/model/DMSDocumentSet.php

```diff
@@ -348,4 +348,56 @@ protected function validate()
         }
         return $result;
     }
+
+    public function canView($member = null)
+    {
+        return $this->getDMSDocumentSetPermissions($member);
+    }
+
+    public function canCreate($member = null)
+    {
+        return  $this->getDMSDocumentSetPermissions($member);
+    }
+
+    public function canEdit($member = null)
+    {
+        return $this->getDMSDocumentSetPermissions($member);
+    }
+
+    public function canDelete($member = null)
+    {
+        return $this->getDMSDocumentSetPermissions($member);
+    }
+
+    /**
+     * Checks if a then given (or logged in) member is either an ADMIN, SITETREE_EDIT_ALL or has access
+     * to the DMSDocumentAdmin module, in which case permissions is granted.
+     *
+     * To extend use the following from within an Extension subclass:
+     *
+     * <code>
+     * public function updateDMSDocumentSetPermissions($result)
+     * {
+     *     // Do something here
+     * }
+     * </code>
+     * @param Member $member
+     * @return bool
+     */
+    public function getDMSDocumentSetPermissions(Member $member = null)
+    {
+        if (!$member || !(is_a($member, 'Member')) || is_numeric($member)) {
+            $member = Member::currentUser();
+        }
+
+        $result = ($member &&
+            Permission::checkMember(
+                $member,
+                array('ADMIN', 'SITETREE_EDIT_ALL', 'CMS_ACCESS_DMSDocumentAdmin')
+            )
+        );
+
+        $this->extend('updateDMSDocumentSetPermissions', $result);
+        return (bool) $result;
```

Comment:
I think it would be better to add extension points to each of the canEdit, canView etc rather than having it here. I know it's more repetition, but it's also more extensible. e.g.

```php
$extended = $this->extendedCan(__FUNCTION__, $member);
if ($extended !== null) {
    return $extended;
}
```

Thoughts?